testsuite/tests/quotation/quotation_typing.ml

```diff
@@ -0,0 +1,254 @@
+(* TEST
+ expect;
+*)
+
+#syntax quotations on
+
+type s0 = <[int]>;;
+[%%expect {|
+type s0 = <[ int ]>
+|}];;
+
+type s1 = <[string]>;;
+[%%expect {|
+type s1 = <[ string ]>
+|}];;
+
+type 'a s2 = <[$'a -> int]> expr;;
+[%%expect {|
+type 'a s2 = <[ $ ('a) -> int ]> expr
+|}];;
+
+type ('a, 'b) s3 = <[$'a -> $'b -> $'a * $'b]> expr;;
+[%%expect {|
+type ('a, 'b) s3 = <[ $ ('a) -> $ ('b) -> $ ('a) * $ ('b) ]> expr
+|}];;
+
+type ('a, 'b, 'c) s4 = <[$'a list -> $'b option -> $'c]> expr;;
+[%%expect {|
+type ('a, 'b, 'c) s4 = <[ $ ('a) list -> $ ('b) option -> $ ('c) ]> expr
+|}];;
+
+type ('a, 'b) s5 = <[$'a -> [`A of 'b]]> expr;;
+[%%expect {|
+Line 32, characters 35-37:
+32 | type ('a, 'b) s5 = <[$'a -> [`A of 'b]]> expr;;
+                                        ^^
+Error: Type variable "b" is used at Line 32, characters 35-37
```

Comment:
I think we don't need to print the location, because it's already pointed at.

Reply:
This doesn't seem to have been fixed. I am still seeing it in error messages.